mlflow/pyfunc/__init__.py

```diff
@@ -3091,6 +3147,16 @@ def predict(self, context, model_input: List[str], params=None) -> List[str]:
         )
 
 
+def update_signature_for_type_hint_from_example(input_example: Any, signature: ModelSignature):
+    if _is_example_valid_for_type_from_example(input_example):
+        signature._type_hint_from_example = True
+    else:
+        _logger.warning(
+            "Input example must be one of pandas.DataFrame, "
```

Comment:
Let's expand with reporting the structure that was provided as an input example to remove a few UX steps of having to look up what that example actually was.